Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.h

```diff
@@ -36,29 +42,56 @@ class SkylanderPortalWindow : public QWidget
   std::array<std::optional<Skylander>, MAX_SKYLANDERS> m_sky_slots;
 
 private:
+  // Window
   void CreateMainWindow();
-  void OnEmulationStateChanged(Core::State state);
-  void CreateSkylander(u8 slot);
-  void ClearSkylander(u8 slot);
-  void EmulatePortal(bool emulate);
-  void LoadSkylander(u8 slot);
-  void LoadSkylanderPath(u8 slot, const QString& path);
-  void UpdateEdits();
+  QVBoxLayout* CreateSlotLayout();
+  QVBoxLayout* CreateFinderLayout();
   void closeEvent(QCloseEvent* bar) override;
   bool eventFilter(QObject* object, QEvent* event) final override;
 
-  QCheckBox* m_checkbox;
-  QGroupBox* m_group_skylanders;
-};
+  // UI
+  void EmulatePortal(bool emulate);
+  void SelectCollectionPath();
+  void LoadSelected();
+  void LoadFromFile();
+  void ClearSlot(u8 slot);
+  void CreateSkylanderAdvanced();
 
-class CreateSkylanderDialog : public QDialog
-{
-  Q_OBJECT
+  // Behind the scenes
+  void OnEmulationStateChanged(Core::State state);
+  void OnCollectionPathChanged();
+  void RefreshList();
+  void UpdateCurrentIDs();
+  void CreateSkyfile(const QString& path, bool loadAfter);
+  void LoadSkyfilePath(u8 slot, const QString& path);
+  void UpdateSlotNames();
 
-public:
-  explicit CreateSkylanderDialog(QWidget* parent);
-  QString GetFilePath() const;
+  // Helpers
+  bool PassesFilter(QString name, u16 id, u16 var);
+  QString GetFilePath(u16 id, u16 var);
+  u8 GetCurrentSlot();
+  int GetElementRadio();
+  QBrush GetBaseColor(std::pair<const u16, const u16> ids);
 
-protected:
-  QString m_file_path;
+  bool m_emulating;
+  QCheckBox* m_enabled_checkbox;
+  QFrame* m_group_skylanders;
+  QFrame* m_command_buttons;
+  QRadioButton* m_slot_radios[16];
```

Comment:
`std::array<QRadioButton*, 16> m_slot_radios;`

Same for `m_game_filters` and `m_element_filter`.